### PR TITLE
Use switch when checking status enum values

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

File: controllers/nova_controller.go [hunk 1]
```diff
@@ -208,12 +209,14 @@ func (r *NovaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (resul
 	for _, cellName := range orderedCellNames {
 		cellTemplate := instance.Spec.CellTemplates[cellName]
 		cellDB, status, err := r.ensureCellDB(ctx, h, instance, cellName, cellTemplate)
-		if err != nil {
+		switch status {
```
Comment (bogdando) [inline controllers/nova_controller.go:212]: this eats the err != nil case handling

File: controllers/nova_controller.go [hunk 2]
```diff
@@ -281,12 +285,14 @@ func (r *NovaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (resul
 			cellMQ, status, err = r.ensureMQ(
 				ctx, h, instance, cellName+"-transport", cellTemplate.CellMessageBusInstance)
 		}
-		if err != nil {
+		switch status {
```
Comment (bogdando) [inline controllers/nova_controller.go:288]: ditto

**Review by bogdando — CHANGES_REQUESTED** [review]

File: controllers/nova_controller.go (hunk 1 above)
Comment (gibizer) [reply controllers/nova_controller.go:212]: `status == DBFailed` and `err != nil` are equivalent based on https://github.com/openstack-k8s-operators/nova-operator/blob/72619e749746e66f54d368615f34fa1bc0e5e2ae/controllers/nova_controller.go#L497-L499

File: controllers/nova_controller.go (hunk 2 above)
Comment (gibizer) [reply controllers/nova_controller.go:288]: ditto :)

**Review by bogdando — APPROVED** [review]

**Pull request merged**